src/Vault.sol

```diff
@@ -0,0 +1,398 @@
+        _deposit(asset_, _msgSender(), receiver, assets_, shares);
+
+        return shares;
+    }
+
+    function paused() public view returns (bool) {
+        return _getVaultStorage().paused;
+    }
+
+    function rateProvider() public view returns (address) {
+        return _getVaultStorage().rateProvider;
+    }
+
+    function bufferStrategy() public view returns (address) {
+        return _getVaultStorage().bufferStrategy;
+    }
+
+    function processAccounting() public {
+        AssetStorage storage assetStorage = _getAssetStorage();
+        StrategyStorage storage strategyStorage = _getStrategyStorage();
+        uint256 totalBaseBalance = 0;
+
+        for (uint256 i = 0; i < assetStorage.list.length; i++) {
+            address asset_ = assetStorage.list[i];
+            uint256 idleBalance = IERC20(asset_).balanceOf(address(this));
+            assetStorage.assets[asset_].idleBalance = idleBalance;
+            totalBaseBalance += _convertAssetToBase(asset_, idleBalance);
+        }
+
+        for (uint256 i = 0; i < strategyStorage.list.length; i++) {
+            address strategy = strategyStorage.list[i];
+            uint256 idleBalance = IERC20(strategy).balanceOf(address(this));
+            strategyStorage.strategies[strategy].idleBalance = idleBalance;
+            totalBaseBalance += _convertAssetToBase(strategy, idleBalance);
+        }
+
+        _getVaultStorage().totalAssets = totalBaseBalance;
+    }
+
+    function processAllocation(address[] calldata targets, uint256[] memory values, bytes[] calldata data)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        for (uint256 i = 0; i < targets.length; i++) {
+            bool success;
+            bytes memory returnData;
+
+            // Determine the type of target and process accordingly
+            if (_getAssetStorage().assets[targets[i]].active) {
+                // Process asset approvals // QUESTION: Assumiung we only need approvals for assets
+                if (data[i].length > 0 && data[i][0] == abi.encodeWithSignature("approve(address,uint256)")[0]) {
+                    (success, returnData) = targets[i].call{value: values[i]}(data[i]);
+                    if (!success) revert ProcessFailed(returnData);
+                }
+            } else if (_getStrategyStorage().strategies[targets[i]].active) {
+                // Process strategy transactions
+                (success, returnData) = targets[i].call{value: values[i]}(data[i]);
+                if (!success) revert ProcessFailed(returnData);
+
+                // Update idle balance for the strategy
+                uint256 idleBalance = IERC20(targets[i]).balanceOf(address(this));
+                _getStrategyStorage().strategies[targets[i]].idleBalance = idleBalance;
+            } else {
+                // Revert if the target is neither an active asset nor an active strategy
+                revert BadStrategy(targets[i]);
+            }
```

Comment:
in addition to strategies i can see the use case for

swapping
wrapping/unwrapping (eg. oeth to WOETH, stETH to wsteth)

 send funds to the buffer strategy
 
 I wonder if we should just have a  generic whitelist of targets that we update.
 
 Additionally, parameters matter a lot. You may whitelist uniswap, but that would still allow swaps to a random shitcoin